src/itemlistformaction.cpp

```diff
@@ -491,10 +491,23 @@ void ItemListFormAction::process_operation(Operation op,
 			if (cfg->get_configvalue_as_bool(
 				    "markfeedread-jumps-to-next-unread")) {
 				process_operation(OP_NEXTUNREAD);
-			} else {
-				LOG(Level::DEBUG,
-					"ItemListFormAction::reset itempos");
-				f->set("itempos", "0");
+			} else { // reposition to first/last item
+				std::string sortorder =
+					cfg->get_configvalue("article-sort-order");
+
+				if (sortorder.find("date-desc") != std::string::npos) {
+					LOG(Level::DEBUG,
+						"ItemListFormAction:: "
+						"reset itempos to last");
+					f->set("itempos",
+						std::to_string(visible_items.size()));
```

Comment:
I think there is an off-by-one error here; we should use `visible_items.size() - 1` to get the index of the last item.

Reply:
OK